### ISSUE TITLE
Add `model_io` module

### ISSUE THREAD

**Issue by Pale-Blue-Dot-97** [Pale-Blue-Dot-97/Minerva#28]
### Issue
`trainer` is currently quite infelxible when it comes to the format it expects samples to be returned from the `dataloader`, parsed to the model and these returns to then be run through the loss function. This is preventing the generalisation to a wide range of models, datasets and learning approaches.

### Solution

- [x] Add a new module called `model_io` to facilitate `trainer` in handling the IO between the `dataloader`, model and loss function.
- [x] Add the first of these methods as the current setup in `trainer` for supervised segmentation models using `torchgeo` data.
